geographically-distributed-postgresql.mdx

```diff
@@ -0,0 +1,178 @@
+
+In the Notion-clone case, we could make the assumption that if a workspace was created in Europe, it probably belongs to a company in Europe, and it will probably be mostly accessed from Europe.
+
+These two conditions are usually easy to satisfy for B2B multi-tenant SaaS applications. For example, Notion and Slack have workspaces as tenants, Github and Vercel have organizations as tenants, and so on.
+
+## Centralized control plane and authentication
+
+In this pattern, the control plane data tables are kept in a single central region. This typically includes the users table, which means the authentication is done from this central region.
+
+On the other hand, the tenant data is distributed to the region where the tenant has the affinity. Because the data is segmented by tenant, all foreign keys point inside the tenant. This means that by connecting to a single region, you can access all the data for a particular tenant.
+
+![Authenticate with the central region](images/geo-postgres/authenticate-to-central-region.png)
+
+It does mean that logging in and signing up is a bit slower because they are served from the central region, but those operations are relatively rare.
+
+A copy of the application stack is deployed in each region. After logging in / signing up, a JWT (or other [similar tokens](https://fly.io/blog/api-tokens-a-tedious-survey/)) are used to authenticate on the data plane. This means that every subsequent request can be sent directly to the region where the tenant data is stored.
+
+Alternatively, the application stack can run on an edge platform (e.g. CloudFlare workers) and therefore automatically close to the user and (hopefully) close to the tenant region. In this case, the client can pass the tenant region as part of the request, and the edge platform can route the request to the correct region.
+
+## Distributing the per tenant data
+
+In our Notion clone example, let's look at how we'd distribute the `pages` table. We're assuming simplified schema for this table, with a page id column, title and URL columns, and the workspace id foreign key. We're also going to add a region column, which will be used to partition the data. This last one is not strictly required, but it will help with partitioning (see below).
+
+![Pages table, grouped by tenant and region](images/geo-postgres/pages-table.png)
+
+We're showing a single table here, but keep in mind that the data is relational with many foreign keys pointing to other tables. Yet, all the foreign keys point inside the tenant, which means that all tables can be distributed in the same way.
+
+We're going to create a `posts` table in each region and store there the data for the tenants that have affinity to that region.
+In our example, we're going to have three Postgres instances in three regions, and each will have a `pages` table with the same schema.
```

Comment:
> In the above, the tables are presented with their schema name

Is the data node schema enforced anywhere by the controller so that they are in sync? If so, maybe we should mention it or add a line about this. It could also be assumed, but it can't hurt to be explicit

Reply:
Ah, I didn't go into schema management, but indeed that's problematic with stock Postgres, but will be covered by our project. I did consider writing about it, but it felt like the blog post is long enough.

Reply:
I think there's room for it if it's just a mention, but also something we could save for launch week